fix: update path-based wikilinks and markdown links when moving notes

logan-mcduffie opened the pull request:
## Summary

- Fixes issue where path-based wikilinks like `[[folder/note]]` were not updated when moving notes
- Adds support for updating markdown-style links like `[text](folder/note.md)`
- Maintains backward compatibility with simple `[[note]]` wikilinks

## Changes

- Added `GenerateLinkReplacements()` function that generates all link patterns to update
- Added `normalizePathSeparators()` helper for cross-platform path handling
- Updated `UpdateLinks()` to use the new comprehensive replacement patterns

## Link Types Now Supported

| Simple wikilink | `[[note]]` |
| Wikilink with alias | `[[note\|alias]]` |
| Wikilink with heading | `[[note#heading]]` |
| Path-based wikilink | `[[folder/note]]` |
| Markdown link | `[text](folder/note.md)` |
| Relative markdown | `[text](./folder/note.md)` |

## Test plan

- [x] Added unit tests for `GenerateLinkReplacements()` covering simple, path-based, and nested paths
- [x] Added integration tests for path-based wikilinks
- [x] Added integration tests for markdown links
- [x] Added integration test for mixed link formats
- [x] All existing tests continue to pass

Fixes #44

logan-mcduffie commented on pkg/obsidian/utils.go:
```diff
@@ -33,6 +39,53 @@ func GenerateNoteLinkTexts(noteName string) [3]string {
+	// Relative markdown link: [text](./folder/note.md)
+	replacements["](./"+oldMd+")"] = "](./" + newMd + ")"
+	replacements["](./"+oldPathNoExt+")"] = "](./" + newPathNoExt + ")"
+
```
> The markdown link replacements are added unconditionally, which could create incorrect mappings when the oldNormalized path equals oldBase (simple filename without path). For example, with inputs "note" and "newNote", this creates duplicate entries: `](note.md)` -> `](newNote.md)` and `](note)` -> `](newNote)`. While not necessarily wrong, it adds redundant entries. More critically, when combined with the filepath.Base bug, this could create incorrect replacements for simple note names.

This is a false positive. The markdown link patterns are not redundant - they are 4 distinct keys in the map:

1. `](oldNote.md)` → `](newNote.md)` - standard link with extension
2. `](oldNote)` → `](newNote)` - standard link without extension
3. `](./oldNote.md)` → `](./newNote.md)` - relative link with extension
4. `](./oldNote)` → `](./newNote)` - relative link without extension

Each pattern represents a valid markdown link format that users might have in their notes:

* Some users write `[text](note.md)`, others write `[text](./note.md)`
* Some include the .md extension, others omit it

Since these are different map keys, there's no duplication. All 4 patterns need to be handled to ensure links are updated correctly regardless of which format was used.

Yakitrak replied:
Thank you for clarifying! I honestly only use the first type.

logan-mcduffie reviewed (commented):
Should be good now @Yakitrak

Yakitrak commented:
@logan-mcduffie thanks! I have resolved some of the comments - are the others not applicate (the missing test cases?)

logan-mcduffie commented:
> @logan-mcduffie thanks! I have resolved some of the comments - are the others not applicate (the missing test cases?)

@Yakitrak Yes, the commit 253d719 addressed all the remaining Copilot feedback. Specifically:

* Test coverage (utils_test.go:90 and :110): Added the missing assertions for relative markdown link patterns (`](./oldNote.md)`, `](./oldNote)`, and `](./folder/oldNote)`)
* Cross-platform bug (utils.go:55): Fixed by using `path.Base()` on normalized paths instead of `filepath.Base()` on raw paths
* Docstring (utils.go:46): Updated to clarify cross-platform path normalization behavior

Should be good to merge.

Yakitrak commented:
Thanks again! I will do a release of this and the fontmatter change soon. I may bundle it with the existing PRs if they can be merged in next week or so.
